fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java

```diff
@@ -149,15 +150,19 @@ public Optional<QueryPredicate> toQueryPredicate(@Nonnull final CorrelationIdent
                 // it seems this is a constant expression, try to evaluate it.
                 return tryBoxSelfAsConstantPredicate();
             }
-        } else if (childrenCount == 2) {
+        }
+        Verify.verify(childrenCount == 2);
+        final var leftChild = it.next();
+        final var rightChild = it.next();
+        if (comparisonType == Comparisons.Type.IN) {
+            final var values = Objects.requireNonNull(((LiteralValue<?>) rightChild).getLiteralValue());
+            return Optional.of(new ValuePredicate(leftChild, new Comparisons.ListComparison(comparisonType, ((List<?>)values))));
```

Comment:
In the spirit of the comment above, I think we should do similar graceful processing here w.r.t. casts.

Reply:
done